### DISCUSSION TITLE
建议优化脚本入口

### DISCUSSION THREAD

**Discussion by NickerGrylls** [the1812/Bilibili-Evolved#4472] [功能建议 / Ideas]
仅我个人而言，脚本配置完后并不需要很频繁地调整，同时由于b站UI的设计风格，完全贴边的“功能”和“设置”按钮看起来有些突兀
即便安装了侧栏偏移和自动隐藏，也难免偶尔的误唤出
我正在使用的最新版Edge117.0.2045.55 (正式版本) (64 位)给标签页都套上了个圆角框框，标签页的左右都与屏幕边缘有微小缝隙，导致光标紧贴左边移动时无法唤出脚本面板，必须手动向右微调（若面板停靠右侧则情况甚至更差）

最终的结果便是：无论是在需不需要调整设置时，操作都不尽优雅

我个人给出两个方案供参考（或可共存？）：
其一，新增通用设置，让用户可指定脚本面板仅在特定页面显示（例如https://account.bilibili.com/account/home https://space.bilibili.com 等）
其有一缺陷是用可能遗忘自己指定的页面

其二，将脚本入口改到最常用页面（例如视频播放页）的二级菜单中，如：
![image](https://github.com/the1812/Bilibili-Evolved/assets/120347871/a52d538b-a454-4475-81b6-6824567f13ec)
或以其他方式将脚本面板嵌入为页面的一部分，例如上图中“稿件投诉”的左侧

**Comment by Geeyun-JY3** [comment]
普通视频播放页下面这行够挤了，新版播放页的这些按钮间距还贼大，功能多开几个就放不下了。
给个我的例子：https://github.com/the1812/Bilibili-Evolved/discussions/4173#discussioncomment-5891703
真的还要再放对脚本来说最重要的两个入口吗？
功能面板也不只是在视频播放页才有用处。

另外你要是用 Tampermonkey，现在是支持通过 Tampermonkey 的菜单唤起这两个面板的：https://github.com/the1812/Bilibili-Evolved/releases/tag/v2.8.3-preview

**Reply by NickerGrylls** [reply]
阁下似乎误会了我的主要述求：可选择性地完全不显示当前形式的脚本面板，至于具体实现方式——无论是否采用我给出的方案——其实不影响
不过感谢告知可以通过Tampermonkey唤醒面板，我刚刚将隐藏侧栏的触发范围改到了最小，试了一下，确实很难再误唤出了，谢谢
但这样曲折的方式果然还是不太优雅

既然有这种唤出方式，我就想到了方案三：给“自动隐藏侧边栏”新可选项：永不显示，这也挺简明直接的

**Reply by Geeyun-JY3** [reply]
我就是觉得你的方案二不太可行所以说一下，述求意思还是明白的。

> 既然有这种唤出方式，我就想到了方案三：给“自动隐藏侧边栏”新可选项：永不显示，这也挺简明直接的

Tampermonkey 这个唤出方式我不太清楚是不是通用的，要是不通用还是得有另外的通用唤醒方法才能这么搞。

**Comment by the1812** [comment]
> 仅我个人而言，脚本配置完后并不需要很频繁地调整，同时由于b站UI的设计风格，完全贴边的“功能”和“设置”按钮看起来有些突兀
即便安装了侧栏偏移和自动隐藏，也难免偶尔的误唤出

结合 Tampermonkey 菜单唤醒的功能, 你可以直接把 `.be-settings` 元素屏蔽掉, 我不打算单独做一个功能.

> 我正在使用的最新版Edge117.0.2045.55 (正式版本) (64 位)给标签页都套上了个圆角框框，标签页的左右都与屏幕边缘有微小缝隙，导致光标紧贴左边移动时无法唤出脚本面板，必须手动向右微调（若面板停靠右侧则情况甚至更差）

这得怪 Edge 乱来, 可以在 `edge://flags/#edge-rounded-containers` 关掉它的圆角皮肤来解决. 屏幕边缘 / 角落是可以鼠标盲操作就能高效到达的位置, 所以我才把入口放在边上.

**Reply by NickerGrylls** [reply]
对flags的强大早有耳闻，但没想到这个都能改，感谢
